motionplan/plan.go

```diff
@@ -307,3 +307,30 @@ func (e *ExecutionState) CurrentInputs() map[string][]referenceframe.Input {
 func (e *ExecutionState) CurrentPoses() map[string]*referenceframe.PoseInFrame {
 	return e.currentPose
 }
+
+// CalculateFrameErrorState takes an ExecutionState and a Frame and calculates the error between the Frame's expected
+// and actual positions.
+func CalculateFrameErrorState(e ExecutionState, frame referenceframe.Frame) (spatialmath.Pose, error) {
+	currentInputs, ok := e.CurrentInputs()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currentPose, ok := e.CurrentPoses()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currPoseInArc, err := frame.Transform(currentInputs)
+	if err != nil {
+		return nil, err
+	}
+	poses, err := e.Plan().Path().GetFramePoses(frame.Name())
+	if err != nil {
+		return nil, err
+	}
+	index := e.Index()
+	if index < 0 || index >= len(poses) {
+		return nil, fmt.Errorf("index %d out of bounds for Path of length %d", index, len(poses))
+	}
+	nominalPose := spatialmath.Compose(poses[index], currPoseInArc)
+	return spatialmath.PoseBetween(nominalPose, currentPose.Pose()), nil
```

Comment:
There's no verification here that these poses are in the same frame.

Reply:
Added a check for this